lib/ansible/modules/cloud/google/gcp_url_map.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/python
+# Copyright 2017 Google Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: gcp_url_map
+version_added: "2.4"
+short_description: Create, Update or Destory a Url_Map.
+description:
+    - Create, Update or Destory a Url_Map. See
+      U(https://cloud.google.com/compute/docs/load-balancing/http/url-map) for an overview.
+      More details on the Url_Map API can be found at
+      U(https://cloud.google.com/compute/docs/reference/latest/urlMaps#resource).
+requirements:
+  - "python >= 2.6"
+  - "google-api-python-client >= 1.6.2"
+  - "google-auth >= 0.9.0"
+  - "google-auth-httplib2 >= 0.0.2"
+notes:
+  - Only supports global Backend Services.
+  - Url_Map tests are not currently supported.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
+  url_map_name:
+    description:
+       - Name of the Url_Map.
+    required: true
+  default_service:
+    description:
+       - Default Backend Service if no host rules match.
+    required: true
+  host_rules:
+    description:
+       - The list of HostRules to use against the URL. Contains
+         host_rules, which is a list of hosts and an associated path_matcher.
+    required: false
+  path_matchers:
+    description:
```

Comment:
For these complex types, feel free to go more verbose about the expected data structure. I see the link at the top for the API (https://cloud.google.com/compute/docs/reference/latest/urlMaps#resource) has really good info on the nested type.